### PR TITLE
Support define symbol as literal

### LINKED ISSUE #48: Can't define symbol as string
Current `shecc` can't define symbol as string, as a result, #47 fail to refactor.

```cpp
#define ARCH_PREDEFIND "__arm__"
```
Above expression is  invalid, later patch will fix it.

### PR DESCRIPTION
**Pull request by eecheng87**
Close #48

### CODE REVIEW

**Comment by eecheng87** [comment]
One of reason fail to bootstrap in #47 was fixed in this patch. 
It seems that there still are some bugs in #47, I'll figure out later.

File: src/defs.h [hunk 1]
```diff
@@ -154,6 +154,7 @@ typedef struct {
 typedef struct {
     char alias[MAX_VAR_LEN];
     char value[MAX_VAR_LEN];
+    int numeric;
```
Comment (jserv) [inline src/defs.h:157]: I just wonder why `numeric` is introduced. Can you eliminate it?

Comment (eecheng87) [reply src/defs.h:157]: Before this patch, `shecc` always views defined symbol as T_numeric. There's chance be T_string such as `#define a "text"`.

Comment (jserv) [reply src/defs.h:157]: I see. However, we can still distinguish from string literals on demand.

Comment (eecheng87) [reply src/defs.h:157]: Yes, we can. If we don't want `numeric` as flag, we need to implement another function to distinguish `char*` from literal or numeric. Is that OK?

Comment (jserv) [reply src/defs.h:157]: > Yes, we can. If we don't want `numeric` as flag, we need to implement another function to distinguish `char*` from literal or numeric. Is that OK?

It sounds better.

**Pull request merged**